docs/proposal-multi-user-approach.md

```diff
@@ -13,11 +12,344 @@ via a UI Server running as the suite owner for UI purposes meaning that, from
+For the purposes of clarity for users, and since we have operation (mutation) level granularity, the simpler method of having two groups, `READ` and `ADMIN`, rather than three. These two access groups can be combined for ease of configuration with `ALL`. Access groups should be entered in caps to avoid confusing e.g. READ with read mutation.
+
+### Example UI Server User Configuration
+
+```python
+c.UIServer.authorisation = {
+    "<user1>": ["READ", "pause", "trigger", "message"], # Specified interactions granted to user1
+    "group:<group1>": ["ALL"],                          # Both READ and ADMIN access to workflows
+                                                        # granted to users in group1
+    "<user2>": ["READ", "ADMIN", "!trigger", "!edit"],  # All READ and ADMIN, except trigger and edit
+                                                        # granted to user2
+    "<user3>": ["READ"]                                 # READ only access
+    "<user4>": ["!READ", "!ADMIN"]                      # user4 has no access to workflows (same as !ALL)
+}
+```
+
+### Example Site Configuration
+
+```python
+c.UIServer.site_authorisation = {
+    "<*>": {                    # For all ui-server owners,
+        "<*>": {                # Any authenticated user
+            "default": "READ",  # Will have default read-only access
+        },
+        "<user1>": {              # user1
+            "default": [
+                "!READ", "!ADMIN" # No privileges for all ui-servers
+            ],                    # owners. Note, could also have configured as !ALL
+        },                        # No limit set, so all ui-server owners
+    },                            # limit is also "!READ", "!ADMIN" for user1
+
+    "<server_owner_1>": {                 # For specific UI Server owner,
+        "<*>": {                          # Any authenticated user
+            "default": "READ",            # Will have default read-only access
+            "limit": ["READ", "ADMIN"]    # server_owner_1 is able to give away
+        },                                # all privileges.
+    },
+    "<server_owner_2>": {       # For specific UI Server owner,
+        "<user2>": {            # Specific user2
+           "limit": "ALL"       # Can only be granted a maximum of ALL (READ and ADMIN) by
+        },                      # server_owner2, default access for user2 falls back to
+                                # standard READ only (if server_owner_2/user2 are
+                                # included in other auth config e.g. the top example),
+                                # or none if not in any other auth config sections.
+
+        "group:<groupA>": {                # group denoted with a `group:`
+            "default": ["READ", "ADMIN"]   # groupA has default READ, ADMIN access to server_owner_2's
+        },                                 # workflows
+    },
+    "group:<grp_of_svr_owners>":{          # Group of users who own UI Servers
```

Comment:
How are these groups to be defined?

Reply:
These are ui-server owner members of system groups. The groups will be returned by an `os.getgrouplist` call.